docs/pages/1-generate-https.md

```diff
@@ -168,205 +140,66 @@ this is pronounced ENGINE-X - it took me only 8 years to know that :P). As a san
-to issue the request for credentials.
-
-```bash
-cd $HOME
-```
-Then, write the csr_details.txt file
-
-```bash
-cat > csr_details.txt <<-EOF
-[req]
-default_bits = 2048
-prompt = no
-default_md = sha256
-req_extensions = req_ext
-distinguished_name = dn
- 
-[ dn ]
-C=US
-ST=$STATE
-L=$COUNTY
-O=End Point
-OU=$DOMAIN
-emailAddress=$EMAIL
-CN = www.$DOMAIN
- 
-[ req_ext ]
-subjectAltName = @alt_names
- 
-[ alt_names ]
-DNS.1 = $DOMAIN
-DNS.2 = www.$DOMAIN
-EOF
-```
-
-This should produce a file `csr_details.txt` in your present working directory
-that we will use later.
-
-
-### Step 2. Install Acme Tiny
-We are going to use [Acme Tiny](https://github.com/diafygi/acme-tiny) to both issue
-and renvew our certificates. Let's install it first. I chose to clone to `/tmp` and
-install to `/opt`, you can obviously mix this up. If you are using a Droplet (or
-a server without python) make sure to install it first:
+Certbot automates certificate generation and renewal. In other words, it makes it really easy to setup SSL.
 
-```bash
-# Python
-sudo apt-get install -y python
 ```
-
-Then install Acme Tiny.
-
-```bash
-sudo mkdir /opt/acme_tiny
-cd /tmp && git clone https://github.com/diafygi/acme-tiny
-sudo mv acme-tiny /opt/acme-tiny/
-sudo chown $USER -R /opt/acme-tiny
+sudo add-apt-repository ppa:certbot/certbot
+sudo apt-get update
+sudo apt-get install python-certbot-nginx
 ```
 
-### Step 3. Create Account Key and Parameters
-This step you only need to do once, so I'm bundling it into one. You need to generate a 
-key for your server, and for the best encryption, a `dhparam.pem` key (it takes a while!).
+### Step 2. Get certificates with certbot
 
-```bash
-# Generate a private account key, if doesn't exist
-if [ ! -f "/etc/ssl/certs/account.key" ]
-   then
-   openssl genrsa 4096 > account.key && sudo mv account.key /etc/ssl/certs
-fi
+Now obtain a certificate by running this command. 
+You can specify multiple hostnames, such as one with "www" like this:
 
-# Add extra security (takes a while!)
-if [ ! -f "/etc/ssl/certs/dhparam.pem" ]
-   then
-   openssl dhparam -out dhparam.pem 4096 && sudo mv dhparam.pem /etc/ssl/certs
-fi
 ```
-
-You shouldn't need to regenerate these files.
-
-### Step 4. Backup Previous Key / Certificate
-This step you only need to do if you have a previously created certificate and domain
-key. It's probably not super necessary, but I do it anyway. The certificate we will make
-(and/or backup) is called `expfactory.cert` and the key is `expfactory.key`.
-
-```bash
-# backup old key and cert
-if [ -f "/etc/ssl/private/domain.key" ]
-   then
-   sudo cp /etc/ssl/private/domain.key{,.bak.$(date +%s)}
-fi
-
-if [ -f "/etc/ssl/certs/chained.pem" ]
-   then
-   sudo cp /etc/ssl/certs/chained.pem{,.bak.$(date +%s)}
-fi
-
-if [ -f "/etc/ssl/certs/domain.csr" ]
-   then
-   sudo cp /etc/ssl/certs/domain.csr{,.bak.$(date +%s)}
-fi
+certbot certonly --nginx -d expfactory.dynu.net -d www.expfactory.dynu.net
 ```
```

Comment:
Is there a way to add an email, for notifications / other?

Reply:
Ah, I forgot to mention this, when certbot first generates a certificate, it asks for the user's email.
If you give it an email, it will send you a reminder at 20 days, 10 days, a 1 days before expiry.
https://letsencrypt.org/docs/expiration-emails/

Reply:
I've added this in.

Reply:
is there a way to do this headless? We would want the user to be able to write a script to do it (e.g., not have any interactive prompts)

Reply:
We could pass a args to cert-bot to take care of the prompts

Instead of

`certbot certonly --nginx -d expfactory.dynu.net -d www.expfactory.dynu.net`

It could be 
`certbot certonly --nginx -d $DOMAIN1 -d $DOMAIN2 --email $EMAIL --agree-tos --redirect`

With those shell vars specified earlier.

Is that all we need?

Reply:
Yes! I like that. Let's take the approach that I was doing before, have the user define these variables once, and then use programmatically (that way they could be input arguments to a script). For example:

```bash
DOMAIN=expfactory.dynu.net
EMAIL=myawesome@email.com
```
```bash
certbot certonly --nginx -d "www.${DOMAIN}" -d "${DOMAIN}" --email "${EMAIL}" --agree-tos --redirect
```
We can assume (for now) that the user is just requesting certificates for the server he/she is sitting on (e.g., we wouldn't have domain1 and domain2.

Reply:
yes that looks great!